service/src/main/kotlin/fi/espoo/evaka/placement/PlacementPlanQueries.kt

```diff
@@ -144,11 +145,25 @@ fun Database.Read.getPlacementPlans(unitId: DaycareId, from: LocalDate?, to: Loc
 SELECT 
     pp.id, pp.unit_id, pp.application_id, pp.type, pp.start_date, pp.end_date, pp.preschool_daycare_start_date, pp.preschool_daycare_end_date,
     pp.unit_confirmation_status, pp.unit_reject_reason, pp.unit_reject_other_reason,
-    p.id as child_id, p.first_name, p.last_name, p.date_of_birth
+    p.id as child_id, p.first_name, p.last_name, p.date_of_birth, (d.status = 'REJECTED'::decision_status AND d.updated > now() - interval '2 week') AS rejected_by_citizen
 FROM placement_plan pp
 LEFT OUTER JOIN application a ON pp.application_id = a.id
 LEFT OUTER JOIN person p ON a.child_id = p.id
-WHERE unit_id = :unitId AND a.status = ANY(:statuses::application_status_type[]) AND deleted = false AND pp.unit_confirmation_status != 'REJECTED'::confirmation_status
+LEFT JOIN decision d ON d.application_id = a.id
+WHERE pp.unit_id = :unitId AND (
+                                (
+                                    a.status = ANY(:statuses::application_status_type[])
+                                    AND a.status != 'REJECTED'::application_status_type
+                                    AND deleted = false
+                                )
+                                OR (
+                                    d.status = 'REJECTED'::decision_status
+                                    AND a.status = ANY(:statuses::application_status_type[])
+                                    AND a.status = 'REJECTED'::application_status_type
```

Comment:
Having two `a.status` predicates here is slightly confusing.
`a.status = ANY(:statuses::application_status_type[])` is common for both sides of `OR` so it should be extracted to top level.

```
(a.status != 'REJECTED'::application_status_type AND deleted = false)
OR
(a.status = 'REJECTED'::application_status_type AND d.status = ...)
```

Reply:
This query likely does full table scans. That might already be the case in production code but adding a timestamp interval filtering on an non-indexed column does not help. At the very least, the performance needs to be monitored after production deployment.

Reply:
Thanks for the feedback @kimifoud. This query is also used in fetching the plans that are being confirmed by the host. The current (confusing) common condition is because `A and (B or C)` is not equal to `(A or B) and (A or C)`.